src/runtime/node/perf_hooks/index.ts

```diff
@@ -0,0 +1,41 @@
+import type perf_hooks from "node:perf_hooks";
+import type { IntervalHistogram } from "node:perf_hooks";
+import * as performanceApis from "../../web/performance/index";
+import { constants } from "./constants";
+
+export { constants } from "./constants";
+export * from "../../web/performance";
+
+const mockRecordableHistogram = () =>
+  Object.create({
+    min: 9_223_372_036_854_776_000,
+    max: 0,
+    mean: null,
+    exceeds: 0,
+    stddev: null,
+    count: 0,
+    percentile: {
```

Comment:
just checked locally not sure why types pass but checking in runtime, `percentile` is a function (PS: there are more)

pushed impl in 3ce186d3ee2b5126f86a9768dc3d87932a370988 in the same PR. feel free to amend or update it.

Reply:
Awesome, thanks! I probably just made a typo and confused `percentile` with `percentiles`. I'm also a little surprised typecheck didn't catch this. 

I pushed a commit that makes a base `Histogram` class that both `IntervalHistogram` and `RecordableHistogram` derive from. Let me know if that works or if there's some consideration for not using a base class.

Also just curious, where does `buckets` come from? I don't see that on either type of histogram in node repl.

Reply:
Thanks & nice find (buckets: 3AM coding and trusting Copilot auto-completion 😆)